decidim-core/vendor/assets/javascripts/form_datepicker.js.es6

```diff
@@ -5,7 +5,7 @@
       const initialDate = $(node).data("startdate") || "";
       const pickTime = $(node).data("timepicker") === "";
       const languageProps = $(node).fdatepicker.dates[language] && $(node).fdatepicker.dates[language].format;
-      let format = languageProps || "mm/dd/yyyy";
+      let format = $(node).data("date-format") || languageProps || "mm/dd/yyyy";
```

Comment:
Wait, this file is vendored, so we shouldn't modify it. Maybe we should port this to the official repo? I think @deivid-rodriguez has some pending PRs there, if we have no answer there maybe we can import the whole file to `decidim-core/app/assets/javascripts/` so we can modify it?

Reply:
Thoughts @deivid-rodriguez ?

Reply:
That, or fork our own version and use it.

Reply:
Ok, lets wait for @deivid-rodriguez feedback's and decide then.

Reply:
Actually, this file is in the vendor folder, but I don't think it's actually vendored. It should be moved to another folder to avoid the confusion.

However, I'm not sure this patch is going to work as we expect. The previous version kept a hidden field in a consistent format (`%Y-%m-%d`) that we know will be properly parsed in the backend. Now we have removed that and we are relying on the date introduced in the frontend to be properly understood in the backend. That will work if the new `date-format` parameter is always the same (and a format properly interpreted in the backend), but in that case we have removed the feature of locales being able to customize the format they use.

Can we make the field manually writable (with a format placeholder, tooltip or something with the expected format), but keep the hidden field so that the value sent to the server is independent from the client's locale?

Reply:
I'd rather remove this hidden field that causes confusion.

Right now dates are automagically parsed by Rectify::Form objects which in turn rely in [virtus](https://github.com/solnic/virtus) (already deprecated) but I can't find any docs on how to configure date formats so I guess they use Date.parse which is heuristic (but I'm not sure). Maybe I'll contact the project and ask for help.

A solution may be to have dates in the Form objects as Strings and then parse them manually inside (start_date_str attribute + start_date accessor that does the parsing).

Reply:
By the way, I've already added a help text with the format. Placeholders are rendered in gray so clear that can't even be seen.

Reply:
Yes, the alternative is to localize the date sent in the backend, according to the request's locale. But it seems harder to implement than the hidden field solution, because the hidden field solution is already implemented :smile:.

Is the removal of the hidden field necessary so that you can make manually introduced dates work? If not, I think the potential refactoring we're discussing now can be done in a separate PR. If yes, then I guess you need to implement that custom date parsing you're suggesting in the backend so that we don't break the ability for locales to customize their date format.

Reply:
By the way, `virtus` coercing seems to be implemented in a separate gem and it does not use `Date.parse` but `to_{date,time,datetime}` activesupport conversion methods. See https://github.com/solnic/coercible/blob/master/lib/coercible/coercer/time_coercions.rb. I don't see an easy way to make those coercions use custom formats, and it seems like an abandoned gem anyways so any potential PRs might get ignored.

So your alternative idea of keeping the attribute in the form object as a string and do the parsing on a reader used by commands objects looks better to me.

I'm still unsure though whether removing the hidden field and moving that logic down to the form object is necessary to fix what we're trying to fix here.

Reply:
Mmm..  now I see where the coercions where... in another gem.
Another option that we have is to take advantage of Ruby's ability to reopen modules and classes, and overwrite `to_date` method in `Coercible::Coercer::TimeCoercions`.
This seems to be the best option to me right now, I'll give it a try.

Reply:
@mrcasals and @deivid-rodriguez 
With [this initializer](https://github.com/decidim/decidim/pull/3724/files#diff-f095e769b525fe7e2d84b17a16d12edd) we're now able to configure Virtus and thus Rectyfy::Form when coercing Strings into Dates. So I guess this review comment is now solved.

Reply:
Nice job! I normally avoid monkeypatching unless I'm convinced it's the only or the best option, and you didn't answer whether we could fix the bug while keeping the current solution, and why / why not. Anyways, it seems good at first sight for virtus to consider localized dates, so let's assume it's the best thing to do here. :+1: Even if the gem seems unmaintained, can you also send a PR there? With that we can:

* Potentially avoid the monkeypatch in the future.
* Potentially get feedback of why what we're doing is a bad idea.
* Confirm whether the gem is maintained or not.

This patch is now a working solution, and I admit removing the hidden field is nice, so let's do it. Ping me when you're done with the patch and CI is green and I'll do another review pass!